src/mesh/mesh_base.C

```diff
@@ -478,45 +476,70 @@ unsigned int MeshBase::recalculate_n_partitions()
 #ifdef LIBMESH_ENABLE_DEPRECATED
 const PointLocatorBase & MeshBase::point_locator () const
 {
-  libmesh_deprecated();
+  auto point_locator_ptr = _point_locator->load(std::memory_order_acquire);
 
-  if (_point_locator.get() == nullptr)
+  // If there's no master point locator, then we need one.
+  if (point_locator_ptr == nullptr)
     {
-      // PointLocator construction may not be safe within threads
-      libmesh_assert(!Threads::in_threads);
+      Threads::spin_mutex::scoped_lock lock(Threads::spin_mtx);
```

Comment:
Is it safe to use Threads::spin_mtx here rather than _point_locator_mutex?  What if this ends up getting called from within other code holding the same global mutex?